src/python/pants/engine/fs.py

```diff
@@ -171,9 +171,10 @@ class DirectoryWithPrefixToAdd:
 
 @dataclass(frozen=True)
 class DirectoryToMaterialize:
-  """A request to materialize the contents of a directory digest at the provided path."""
-  path: str
+  """A request to materialize the contents of a directory digest at the provided path,
```

Comment:
Adding a bit of explanation of the behavior of `path` in another sentence below this one would be good: strawman: "If path is set, it is prefixed on the paths contained within the Snapshot".

Reply:
What do you think about renaming `path` to be `prefix_path`, `path_relative_to_buildroot`, or `prefix_path_relative_to_buildroot`?

I would like `prefix_path` the most. I was quite confused when adding Workspace to `fmt.py`  what `path` meant. What confused me is that the directory digest already has paths in it, like `src/python/pants`, and I thought `path` was suggesting that I need to somehow specify every single path used like `src/python/pants` etc. Until I realized, it just means a prefix.

Reply:
`prefix_path` or `path_prefix` would be good, yea.